### ISSUE TITLE
Docs issue, multiple italics in doc paragraph cause mega-italicized block

### ISSUE THREAD

**Issue by rlmark** [unisonweb/unison#3895]
The following doc fragment with underscores around words for emphasis will instead italicize the entire block between the desired emphasized words. 

```
underscoreIssueDoc = {{

Blah blah  _namespaces_. blah blah blah blah _another word_ blah blah.

Another paragraph with _underscores_ is ok.
}}
```

In the ucm this term is saved as: 

```
  underscoreIssueDoc : Doc2
  underscoreIssueDoc =
    {{
    Blah blah
    ___namespaces__. blah blah blah blah__another word___ blah
    blah.
    
    Another paragraph with __underscores__ is ok.
    }}
```

Transcript with this code is attached here: 
[underscore.md](https://github.com/unisonweb/unison/files/11151097/underscore.md)

**Comment by sellout** [comment]
tl;dr: Closing delimiters need to be followed by whitespace. For a workaround, add a space before the period in `_namespaces_ .`

It’s somewhat apparent from the re-generated `Doc` in the report, but
<details>
<summary>the parsed structure is</summary>

```haskell
UntitledSection
  [ Paragraph
      [ Word "Blah"
      , Word "blah"
      , Italic
          (Paragraph
            [ Doc.Group
                (Doc.Join
                  [ Word "namespaces"
                  , Doc.Group
                      (Doc.Join
                        [ Italic
                            (Paragraph
                              [ Word "."
                              , Word "blah"
                              , Word "blah"
                              , Word "blah"
                              , Word "blah"
                              ])
                        , Word "another"
                        ])
                  ])
            , Word "word"
            ])
      , Word "blah"
      , Word "blah."
      ]
  , Paragraph
      [ Word "Another"
      , Word "paragraph"
      , Word "with"
      , Italic (Paragraph [Word "underscores"])
      , Word "is"
      , Word "ok."
      ]
  ]
```
</details>
You can see that the italic goes from the first delimiter to the last, and a nested italic goes from the second delimiter (intended to terminate the first italic) to the third (intended to start a second italic).

In the parser, this is because we need whitespace after the closing delimiter (but not before an opening delimiter). So `_namespaces_.` starts with an opener, but doesn’t end with a closer (and `_another` doesn’t have a closer either).

<details>
<summary>Adding a space in `_namespaces_ .` gives us what we want.</summary>

```haskell
UntitledSection
  [ Paragraph
      [ Word "Blah"
      , Word "blah"
      , Italic (Paragraph [Word "namespaces"])
      , Word "."
      , Word "blah"
      , Word "blah"
      , Word "blah"
      , Word "blah"
      , Italic (Paragraph [Word "another", Word "word"])
      , Word "blah"
      , Word "blah."
      ]
  , Paragraph
      [ Word "Another"
      , Word "paragraph"
      , Word "with"
      , Italic (Paragraph [Word "underscores"])
      , Word "is"
      , Word "ok."
      ]
  ]
```
</details>

I think the ending delimiter should be a bit more flexible, allowing “non-word” characters, not simply whitespace.

**Comment by sellout** [comment]
This is a duplicate of #2890.
